### PR TITLE
Refactor metrics

### PR DESCRIPTION
**Pull request by poluektik**
Hi @gpertzov, your last comments were applied.

**Few words about the changes**

1.Aletheia’s API will be changed to work with MetricFactoryProvder instead of MetricFactory, this way the 
implementation of the Provider will be done on Implementation side, it will be easier to refactor metrics next time without changing the core library

2. Added labels to metrics  interfaces, to align project with Prometheus, added dependency to swinfra metrics lib

3. Names of the counters  aligned with prometheus naming convention

4. labels enables us to reduce the need in different sorts of factory metric providers, one interface is enough

### CODE REVIEW

File: aletheia-core/src/main/java/com/outbrain/aletheia/DatumProducerBuilder.java [hunk 1]
```diff
@@ -50,50 +46,47 @@ private DatumProducerBuilder(final Class<TDomainClass> domainClass) {
...
-                productionEndPointInfo.getProductionEndPoint(),
-                datumProducerConfig);
+            productionEndPointInfo.getProductionEndPoint(),
+            datumProducerConfig);
 
     final BreadcrumbDispatcher<TDomainClass> datumAuditor;
-    final MetricFactoryProvider metricFactoryProvider;
+    final boolean isBreadcrumbs = domainClass.equals(Breadcrumb.class);
 
     if (!domainClass.equals(Breadcrumb.class)) {
```
Comment (gpertzov) [inline aletheia-core/src/main/java/com/outbrain/aletheia/DatumProducerBuilder.java]: Use isBreadcrumbs

File: aletheia-core/src/main/java/com/outbrain/aletheia/DatumConsumerStreamsBuilder.java [hunk 2]
```diff
@@ -76,22 +72,20 @@ private DatumConsumerStreamsBuilder(final Class<TDomainClass> domainClass) {
             datumConsumerStreamConfig);
 
     final BreadcrumbDispatcher<TDomainClass> datumAuditor;
-    final MetricFactoryProvider metricFactoryProvider = new DefaultMetricFactoryProvider(DatumUtils.getDatumTypeId(domainClass),
-            DATUM_STREAM,
-            getMetricsFactory());
+
```
Comment (gpertzov) [inline aletheia-core/src/main/java/com/outbrain/aletheia/DatumConsumerStreamsBuilder.java]: Remove empty lines

File: aletheia-core/src/main/java/com/outbrain/aletheia/DatumEnvelopeStreamsBuilder.java [hunk 3]
```diff
@@ -37,8 +35,9 @@
  */
 public class DatumEnvelopeStreamsBuilder extends BaseAletheiaBuilder {
 
+  @SuppressWarnings("unused")
+  public static final String ENVELOPE_CONSUMER_STREAM = "EnvelopeConsumerStream";
```
Comment (gpertzov) [inline aletheia-core/src/main/java/com/outbrain/aletheia/DatumEnvelopeStreamsBuilder.java]: Remove field if unused

File: aletheia-core/src/main/java/com/outbrain/aletheia/datum/consumption/openers/BaseEnvelopeOpener.java [hunk 4]
```diff
@@ -34,12 +34,14 @@ public BaseEnvelopeOpener(final BreadcrumbDispatcher<TDomainClass> datumAuditor,
...
-            true);
-    metricFactory.createGauge("Timestamp.Logical", "DelayAverageInSeconds", logicalDelayAverager);
-    metricFactory.createGauge("Timestamp.Logical", "DelayAverageInMillis", logicalDelayInMillisAverager);
+    futureLogicalMessagesCount = metricFactory.createCounter("timestampLogicalFromTheFuture", "future logical message count");
+    logicalTimestampDelayHistogram = metricFactory.createHistogram(
+            "Timestamp_Logical_DelayHistogramInSeconds",
+            "delay in seconds",
+            new double[]{.001, .005, .01, .05, 0.1, 0.5, 1, 5, 10});
+
+    metricFactory.createGauge("timestampLogicalDelayAverageInSeconds", "Breadcrumbs dispatcher average delay in seconds", logicalDelayAverager);
```
Comment (gpertzov) [inline aletheia-core/src/main/java/com/outbrain/aletheia/datum/consumption/openers/BaseEnvelopeOpener.java]: This is relevant to every datum, not specifically Breadcrumbs

File: aletheia-core/src/main/java/com/outbrain/aletheia/datum/consumption/openers/BaseEnvelopeOpener.java [hunk 5]
```diff
@@ -34,12 +34,14 @@ public BaseEnvelopeOpener(final BreadcrumbDispatcher<TDomainClass> datumAuditor,
...
+    metricFactory.createGauge("timestampLogicalDelayAverageInMillis", "Breadcrumbs dispatcher average delay in millis", logicalDelayInMillisAverager);
```
Comment (gpertzov) [inline aletheia-core/src/main/java/com/outbrain/aletheia/datum/consumption/openers/BaseEnvelopeOpener.java]: This is the consumption delay of the datum, not related to Breadcrumbs

File: aletheia-core/src/main/java/com/outbrain/aletheia/DatumProducerBuilder.java (hunk 1 above)
Comment (poluektik) [reply aletheia-core/src/main/java/com/outbrain/aletheia/DatumProducerBuilder.java]: 🤦‍♂️

**Comment by poluektik** [comment]
Hi @gpertzov I've changed the code.
please merge.

**Pull request merged**